cmd/queue/main.go

```diff
@@ -299,8 +299,18 @@ func buildServer(ctx context.Context, env config, healthState *health.State, rp
 	// Note: innermost handlers are specified first, ie. the last handler in the chain will be executed first.
 	var composedHandler http.Handler = httpProxy
 	if concurrencyStateEnabled {
-		logger.Info("Concurrency state endpoint set, tracking request counts")
-		composedHandler = queue.ConcurrencyStateHandler(logger, composedHandler, nil, nil)
+		logger.Info("Concurrency state endpoint set, tracking request counts, using endpoint: ", env.ConcurrencyStateEndpoint)
+		pauseMsg, err := queue.CreateConcurrencyStateMessageBody(queue.ConcurrencyStateMessageBody{Action: "pause"})
+		if err != nil {
+			logger.Errorf("unable to create concurrency state pause message body: %s", err)
+		}
+		resumeMsg, err := queue.CreateConcurrencyStateMessageBody(queue.ConcurrencyStateMessageBody{Action: "resume"})
+		if err != nil {
+			logger.Errorf("unable to create concurrency state resume message body: %s", err)
+		}
```

Comment:
This seems quite redundant. Why not pass a `ConcurrencyStateMessageBody` to `ConcurrencyStateRequest` and marshal in the function before returning the closure? Seems better API design too, since `ConcurrencyStateRequest` taking `[]byte` is a bit more generic than seems healthy.

Reply:
I originally was passing a `ConcurrencyStateMessageBody` (though without a separate function for marshalling), and for some reason then the `resume` call would only work once (on any subsequent call, it was throwing a `400` error). Now, however, using `CreateConcurrencyStateMessageBody` within `ConcurrencyStateRequest` works fine. :shrug: Updating now.